Fix optimum-cli command for VLM example in README

helena-intel opened the pull request:
With the existing command users get an error: Channel size 4304 should be divisible by size of group 128.

AlexKoff88 commented on README.md:
```diff
@@ -112,7 +112,7 @@ For more examples check out our [LLM Inference Guide](https://docs.openvino.ai/2
 
 #(Recommended) Same as above but with compression: language model is compressed to int4, other model components are compressed to int8
-optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 MiniCPM-V-2_6
+optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 --group-size 16 MiniCPM-V-2_6
```
The minimum group size we committed to support is 32. @helena-intel, can it work with group size of 32, or 64? The higher is better for performance and model footprint.

helena-intel replied:
Divisors of channel size 4304 are 1, 2, 4, 8, 16, 269, 538, 1076, 2152, 4304 so 32 and 64 don't work either. I tried channel-wise quantization (group size -1), but with the example script the output was empty (the same script worked fine with group size 16). If group size 16 is not supported, should we remove the INT4 recommendation?

AlexKoff88 replied:
Thanks @helena-intel. Actually, the issue is related to the fact that we compress Vision model to int4 as well which is unexpected. We will prepare a fix for Optimum-Intel. But I would use `int8` for now.

AlexKoff88 replied:
As an alternative you can use dataset `--weight-format int4 --dataset contextual`. It works on my end.

helena-intel replied:
Thank you, that is great!

AlexKoff88 commented on README.md:
```diff
@@ -112,7 +112,7 @@ For more examples check out our [LLM Inference Guide](https://docs.openvino.ai/2
 
 #(Recommended) Same as above but with compression: language model is compressed to int4, other model components are compressed to int8
-optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 MiniCPM-V-2_6
+optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 --group-size 16 MiniCPM-V-2_6
```
```suggestion
optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 --dataset contextual MiniCPM-V-2_6 
```

nikita-savelyevv commented on README.md:
```diff
@@ -112,7 +112,7 @@ For more examples check out our [LLM Inference Guide](https://docs.openvino.ai/2
 
 #(Recommended) Same as above but with compression: language model is compressed to int4, other model components are compressed to int8
-optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 MiniCPM-V-2_6
+optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 --dataset contextual MiniCPM-V-2_6 
```
```suggestion
optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 --dataset contextual --num-samples 1 MiniCPM-V-2_6 
```

This is faster and more memory efficient. Since no data-aware method is selected, dataset size won't have effect on the resulting model.

Btw, I plan to publish a fix to optimum-intel some time today.

helena-intel commented:
@AlexKoff88 @nikita-savelyevv on Xeon the method with --dataset, both with and without num-samples, gives an empty result with the README example. With --group-size 16 I do get a good result. Also tried to test on laptop yesterday, but there the model export did not work (unrelated to this).

ilya-lavrenov commented:
build_jenkins

helena-intel commented:
Sor now the commands suggested by Alexander and Nikita don't work with GenAI (see comments and linked issue). There's a ticket about that. Can anyone suggest another model that works well? Smaller is better. Otherwise I'll remove the INT4 suggestion for now, until the issue is fixed in GenAI.

nikita-savelyevv commented:
> For now the commands suggested by Alexander and Nikita don't work with GenAI (see comments and linked issue). There's a ticket about that. Can anyone suggest another model that works well? Smaller is better. Otherwise I'll remove the INT4 suggestion for now, until the issue is fixed in GenAI.

`OpenGVLab/InternVL2-1B` is quite small and works with VLMPipeline. But I would recommend to wait for the fix https://github.com/huggingface/optimum-intel/pull/1058 to be merged.

helena-intel commented:
> `OpenGVLab/InternVL2-1B` is quite small and works with VLMPipeline. But I would recommend to wait for the fix [huggingface/optimum-intel#1058](https://github.com/huggingface/optimum-intel/pull/1058) to be merged.

Thank you! That's a much smaller model, much more user friendly, especially on AI PC. I tested and it works well out of the box with GenAI with the example from the README (after installing einops and timm). I'll update the PR with this. If someone disagrees, let me know.

The fix in optimum-intel is not enough for this because then exporting will work, but inference with GenAI doesn't. When that is fixed in an upcoming GenAI release we could change back to MiniCPM - though I see no downside for having smaller models in README examples.

helena-intel commented:
Changed README to use `OpenGVLab/InternVL2-1B` with INT4. With FP16/FP32 model export, inference with the README example is empty. I added it to the ticket about empty output with MiniCPM. For now, this model with INT4 works, is quick to download and convert, and it's just for an inference example in the README, INT4 should be fine.
